### ISSUE TITLE
没有xtuner.parallel: 'No module named 'xtuner.parallel''

### ISSUE THREAD

**Issue by menkeyi001** [InternLM/xtuner#539]
```shell
ubuntu22.04
安装步骤
conda create --name xtuner-env python=3.10 -y
 conda activate xtuner-env
pip install -U 'xtuner[deepspeed]' -i https://pypi.tuna.tsinghua.edu.cn/simple/

运行
 xtuner train ./internlm2_7b_qlora_oasst1_512_e3_copy.py  --deepspeed deepspeed_zero2
/root/anaconda3/envs/xtuner-env/lib/python3.10/site-packages/torch/cuda/__init__.py:138: UserWarning: CUDA initialization: CUDA driver initialization failed, you might not have a CUDA gpu. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)
  return torch._C._cuda_getDeviceCount() > 0
/root/anaconda3/envs/xtuner-env/lib/python3.10/site-packages/bitsandbytes/cextension.py:31: UserWarning: The installed version of bitsandbytes was compiled without GPU support. 8-bit optimizers, 8-bit multiplication, and GPU quantization are unavailable.
  warn("The installed version of bitsandbytes was compiled without GPU support. "
/root/anaconda3/envs/xtuner-env/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so: undefined symbol: cadam32bit_grad_fp32
[2024-04-02 11:05:56,914] [INFO] [real_accelerator.py:191:get_accelerator] Setting ds_accelerator to cuda (auto detect)
04/02 11:05:58 - mmengine - WARNING - WARNING: command error: 'No module named 'xtuner.parallel''!
04/02 11:05:58 - mmengine - WARNING - 
    Arguments received: ['xtuner', 'train', './internlm2_7b_qlora_oasst1_512_e3_copy.py', '--deepspeed', 'deepspeed_zero2']. xtuner commands use the following syntax:

        xtuner MODE MODE_ARGS ARGS

        Where   MODE (required) is one of ('list-cfg', 'copy-cfg', 'log-dataset', 'check-custom-dataset', 'train', 'test', 'chat', 'convert', 'preprocess', 'mmbench', 'eval_refcoco')
                MODE_ARG (optional) is the argument for specific mode
                ARGS (optional) are the arguments for specific command

    Some usages for xtuner commands: (See more by using -h for specific command!)
```

**Comment by LZHgrla** [comment]
Hi! @menkeyi001 
So sorry for that. This is a known issue about xtuner v0.1.16, and we are working on it!

Before being fixed, you can degrade xtuner to 0.1.15, or install xtuner from source, to resolve this bug.

```
# Approach 1: degrade
pip install xtuner==0.1.15
# Approach 2: install from source
git clone https://github.com/InternLM/xtuner.git
cd xtuner
pip install -e .
```

**Comment by LZHgrla** [comment]
@menkeyi001 
We have fixed this bug.

You can install xtuner>=0.1.17 to solve this issue.
```
pip install 'xtuner>=0.1.17'
```
